### PR TITLE
CBL-4101: adapting tests based on SG 3.0 of walrus mode to also work …

### PR DESCRIPTION
**Pull request by jianminzhao**
…with SG 3.1 backed by CB server.

When Lithium (aka v3.0) was released, all replicator tests were run against SG of the Walrus mode. With v3.1, we started to use SG backed up by actual CB servers. We ported tests in release/3.1, and the associated test framework, to lithium and put them in ReplicatorSGTest, and moved original tests based on Walrus to ReplicatorWalrusTest. This commit adapts the original tests of Walrus to also work with SG(3.1) backed by CB server.

### CODE REVIEW

**Comment by jianminzhao** [comment]
This PR only affects sources in test folder, Replicator/tests.

File: Replicator/tests/ReplicatorWalrusTest.cc [hunk 1]
```diff
@@ -79,11 +98,65 @@ class ReplicatorWalrusTest : public ReplicatorAPITest {
...
     }
 
-};
+    enum AuthType {
+        kAuthNone,
+        kAuthBody,
+        kAuthHeader
+    };
 
+    void notWalrus(AuthType authType =kAuthNone) {
```
Comment (borrrden) [inline Replicator/tests/ReplicatorWalrusTest.cc:107]: This is pretty weak function naming

**Review by borrrden — APPROVED** [review]

File: Replicator/tests/ReplicatorWalrusTest.cc (hunk 1 above)
Comment (jianminzhao) [reply Replicator/tests/ReplicatorWalrusTest.cc:107]: Totally agreed. It's function is to make up the difference that, for non-walrus, we have to supply credentials, whereas in the walrus mode, database "scratch" is in the guest mode.
What about, "withAuthentication(AuthType)" ?

Comment (jianminzhao) [reply Replicator/tests/ReplicatorWalrusTest.cc]: Changed function name to withServerBackedSG()

**Pull request merged**